File: Velero.py
from embarcacion import Embarcacion

class Velero(Embarcacion):
    def __init__(self, nombre, capacidad, longitud, num_velas):
        super().__init__(nombre, "Velero", capacidad, longitud)
        self.num_velas = num_velas

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Número de velas: {self.num_velas}")

    def desplegar_velas(self):
        print(f"El velero {self.nombre} está desplegando {self.num_velas} velas.")

File: Velero.cpp
#include "Velero.h"

Velero::Velero(string n, int c, double l, int a, int v, string m)
    : Embarcacion(n, c, l, a) {
    numVelas = v;
    materialMastil = m;
    cout << "Velero creado: " << nombre << endl;
}

Velero::~Velero() {
    cout << "Velero destruido: " << nombre << endl;
}

void Velero::izarVelas() {
    cout << nombre << " ha izado sus velas." << endl;
}

void Velero::mostrarDatos() {
    cout << "\n=== DATOS DEL VELERO ===" << endl;
    Embarcacion::mostrarDatos();
    cout << "Número de velas: " << numVelas << endl;
    cout << "Material del mástil: " << materialMastil << endl;
}
